### PR TITLE
snakemake and py-snakemake-*: updates to latest versions

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds `snakemake`, v8.25.2, and updates the plugins to their latest versions.

Needs:
- [x] #47525
- [x] #47528

### CODE REVIEW

**Comment by wdconinc** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py:18]: Version `1.1.0` added a requirement

```python
snakemake = "^8.20.5"
```

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py:23]: Version `0.10.1` added a requirement

```python
snakemake = "^8.20.0"
```

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py:19]: Version `0.1.4` adds constraint

```python
snakemake-interface-executor-plugins = "^9.0.0"
```

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py:18]: Version `0.1.5` adds constraint

```python
snakemake = "^8.20.1"
```

File: var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py:18]: I was wondering how I missed that. This is a `dev` dependency, not a regular dependency. https://github.com/snakemake/snakemake-interface-report-plugins/blob/main/pyproject.toml#L14

Do you want to add these too?

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py:23]: Same as above. https://github.com/snakemake/snakemake-executor-plugin-slurm/blob/main/pyproject.toml#L28

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-tes/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-tes/package.py]: Constraints are considered together, so this line needs `when="@:0.1.2"`.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-kubernetes/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-kubernetes/package.py]: This line is missing `when="@:0.1.4"`.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-kubernetes/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-kubernetes/package.py:29]: Version `0.1.5` relaxes following constraint

```diff
- kubernetes = "^27.2.0"
+ kubernetes = ">=27.2.0,<30"
```

Version `0.2.0` adds following constraints

```python
snakemake-interface-common = "^1.17.3"`
```

Version `0.2.1` relaxes following constraint

```diff
- kubernetes = ">=27.2.0,<30"
+ kubernetes = ">=27.2.0,<31"
```

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-googlebatch/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-googlebatch/package.py]: This line is missing `when="@:0.3.0"`.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-googlebatch/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-googlebatch/package.py:20]: Version `@0.5.0` adds following constraint

```python
snakemake = "^8.18.0"
```

The constraint is in dev dependencies, but this plugin needs Snakemake, so it might be better to add it.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-flux/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-flux/package.py]: This line is missing `when="@:0.1.0"`.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-azure-batch/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-azure-batch/package.py]: This line is missing `when="@:0.1.3"`.

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/snakemake/package.py]: It becomes a legacy dependency, we usually push these to the end of package definition.

File: var/spack/repos/builtin/packages/snakemake/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/snakemake/package.py]: This makes it a legacy dependency, we usually push these to the end of package definition.

File: var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-snakemake-interface-report-plugins/package.py:18]: You are right.

It is a bit confusing. On one hand these plugins only do anything if `py-snakemake` is installed. On the other we didn't add Snakemake to dependencies so far.

Snakemake version is resolved indirectly through Snakemake interface packages. Let's make it consistent with how it was and ignore my comment about Snakemake constraint.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-snakemake-executor-plugin-slurm/package.py:23]: Ok, let's skip it.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-azure-batch/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-azure-batch/package.py]: `when="@:0.1"` here is correct but unnecessary. It can stay or be removed for consistency.

File: var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py
Comment (w8jcik) [inline var/spack/repos/builtin/packages/py-snakemake-executor-plugin-drmaa/package.py]: One more missing `when`.

**Review by w8jcik — APPROVED** [review]

**Comment by wdconinc** [comment]
@spackbot run pipeline

**Comment by wdconinc** [comment]
@marcusboden @w8jcik This PR is ready for a final review.

**Review by w8jcik — APPROVED** [review]

**Comment by w8jcik** [comment]
Hello @wdconinc, I already reviewed and approved your changes. I approved it now again if it helps.

Regarding the other maintainer: I would not count on a reply. In my previous pull request it didn't happen for 3 months.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@wdconinc Feel free to merge if you're not waiting for further comments

**Pull request merged**
